oops/oops.go

```diff
@@ -7,8 +7,41 @@ import (
 	"runtime"
 	"strconv"
 	"strings"
+	"sync/atomic"
 )
 
+// filePrefixesToShortCircuit is a set of prefixes of files which, when encountered when serializing
+// an oops error, will skip appending the current frame and all subsequent frames in that stack.
+// It is useful for truncating long call stacks to make stack traces more concise.
+var filePrefixesToShortCircuit atomic.Value // map[string]struct{}{}
+
+func init() {
+	// Make sure we don't panic if we invoke this package and haven't called `SetPrefixesToShortCircuit`.
+	SetPrefixesToShortCircuit()
+}
+
+// SetPrefixesToShortCircuit sets `filePrefixesToShortCircuit` to `prefixes`. While it is thread-safe, it is the
+// caller's responsibility to ensure that there are not multiple callers in the same program calling
+// `SetPrefixesToShortCircuit` concurrently, as this may cause unexpected behavior where different prefixes are
+// filtered from stacktraces based on the set of values passed by the most recent caller of this function.
+func SetPrefixesToShortCircuit(prefixes ...string) {
+	prefixMap := make(map[string]struct{})
+	for _, prefix := range prefixes {
+		prefixMap[prefix] = struct{}{}
+	}
+	filePrefixesToShortCircuit.Store(prefixMap)
+}
+
+// GetPrefixesToShortCircuit returns the current set of file prefixes to short-circuit.
+func GetPrefixesToShortCircuit() []string {
```

Comment:
Do you think it is worth calling out order is non-deterministic? We could also sort here if you want.

Reply:
I'll call out and let callers decide what they want to do with the output.